Examples/Framework/include/ActsExamples/Framework/WhiteBoard.hpp

```diff
@@ -95,21 +100,42 @@ inline void ActsExamples::WhiteBoard::add(const std::string& name, T&& object) {
 
 template <typename T>
 inline const T& ActsExamples::WhiteBoard::get(const std::string& name) const {
-  auto it = m_store.find(name);
-  if (it == m_store.end()) {
+  if (!exists(name)) {
     throw std::out_of_range("Object '" + name + "' does not exists");
   }
-  const IHolder* holder = it->second.get();
+  const std::string& resolvedName = resolve(name);
+  const IHolder* holder = m_store.at(resolvedName).get();
 
   const auto* castedHolder = dynamic_cast<const HolderT<T>*>(holder);
   if (castedHolder == nullptr) {
-    throw std::out_of_range("Type mismatch for object '" + name + "'");
+    throw std::out_of_range("Type mismatch for object '" + resolvedName + "'");
   }
 
-  ACTS_VERBOSE("Retrieved object '" << name << "'");
+  ACTS_VERBOSE("Retrieved object '" << resolvedName << "'");
   return castedHolder->value;
 }
 
 inline bool ActsExamples::WhiteBoard::exists(const std::string& name) const {
-  return m_store.find(name) != m_store.end();
-}
\ No newline at end of file
+  return (m_store.find(name) != m_store.end()) ||
+         (m_alias.find(name) != m_alias.end());
```

Comment:
This would also have to do the resolution to be correct no? I.e. if an alias exists, but the value the alias points at does not, this function should return false.

Reply:
Ah never mind, when you set an alias it checks if the underlying key exists.